### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jlovejoy**
fix a couple links that were outdated, no corresponding issue

@swinslow can you check and merge?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -4,15 +4,15 @@ Please familiarize yourself with the SPDX License List and its supporting docume
...
 
-* [SPDX License List Matching Guidelines](https://spdx.org/spdx-license-list/matching-guidelines) provides guidelines to be used for the purposes of matching licenses and license exceptions against those included on the SPDX License List.
+* [SPDX License List Matching Guidelines]([https://spdx.org/spdx-license-list/matching-guidelines](https://spdx.github.io/spdx-spec/v2.3/license-matching-guidelines-and-templates/)) provides guidelines to be used for the purposes of matching licenses and license exceptions against those included on the SPDX License List.
 
-* [SPDX Specification](https://spdx.org/specifications): It is helpful to be familiar with certain sections of the SPDX Specification that use or deal with the SPDX License List. In particular: sub-sections related to license information in Section 7, 8, and 10; Appendices B, D, and E.
+* [SPDX Specification]([https://spdx.org/specifications](https://spdx.dev/use/specifications/)): It is helpful to be familiar with certain sections of the SPDX Specification that use or deal with the SPDX License List. In particular: sub-sections related to license information in Section 7, 8, and 10; Appendices B, D, and E.
```
Comment (xsuchy) [inline CONTRIBUTING.md:9]: This does not make a sense. You added link to a link. This part should be reverted.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -4,15 +4,15 @@ Please familiarize yourself with the SPDX License List and its supporting docume
 
 * [Explanation of the fields contained on the list](DOCS/license-fields.md)
 
-* [SPDX License List Matching Guidelines](https://spdx.org/spdx-license-list/matching-guidelines) provides guidelines to be used for the purposes of matching licenses and license exceptions against those included on the SPDX License List.
+* [SPDX License List Matching Guidelines]([https://spdx.org/spdx-license-list/matching-guidelines](https://spdx.github.io/spdx-spec/v2.3/license-matching-guidelines-and-templates/)) provides guidelines to be used for the purposes of matching licenses and license exceptions against those included on the SPDX License List.
```
Comment (xsuchy) [inline CONTRIBUTING.md:7]: You added link to link. That does not have sense. The old link is dead so it should be updated - that is correct. But if we keep the version we will have to update it every second year. You can use https://spdx.github.io/spdx-spec/latest/license-matching-guidelines-and-templates/  which redirect to latest tag. Right now to v2.3. That should work in future without need for a change.

File: CONTRIBUTING.md (hunk 1 above)
Comment (jlovejoy) [reply CONTRIBUTING.md]: fixed, thanks!

File: CONTRIBUTING.md (hunk 2 above)
Comment (jlovejoy) [reply CONTRIBUTING.md]: fixed now

**Review by swinslow — COMMENTED** [review]
Reviewing after the fact, but LGTM

**Pull request merged**